wandb/sdk/internal/incremental_table_util.py

```diff
@@ -1,23 +1,59 @@
+from __future__ import annotations
+
 import time
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 import wandb
 
 if TYPE_CHECKING:
     from wandb import Table
     from wandb.sdk.artifacts.artifact import Artifact
+
     from ..wandb_run import Run as LocalRun
 
 ART_TYPE = "wandb-run-incremental-table"
 
 
-def init_artifact(run: "LocalRun", sanitized_key: str) -> "Artifact":
+def handle_resumed_run(incr_table: Table, run: LocalRun, key: str):
+    """Handle different scenarios when a run is resumed.
+
+    Check the summary to see if there was an incremental table that was logged for
+    the key and assign the previous_increments_paths and increment_num to the table.
```

Comment:
The name and docstring summary line describe when the function is called (when a table is used in a run that was resumed), but should instead describe what it does. On the other hand, the description will likely go stale as the function is updated, like if the `previous_increment_paths` and `increment_num` fields are renamed, or if additional actions need to be done.